### DISCUSSION TITLE
NextAuth Login Flow Diagram

### DISCUSSION THREAD

**Discussion by BetweenTwoTests** [nextauthjs/next-auth#4736] [Help]
I am trying to document the login flow for NextAuth. As far as I understand it, the following diagram is roughly how it the flow goes. I'm looking for feedback from other NextAuth users to see if my understanding is correct.

Link: https://1drv.ms/u/s!ApO_Ig_VxP_pfAiZuQa9_lumyKQ?e=4rILn4 

![NextAuthLoginProcess](https://user-images.githubusercontent.com/10789273/174686366-6b4bc7f9-92f1-4880-b382-facd729d7033.jpg)

Also, if you have suggestions for tools to draw similar diagram, I am open to suggestions. I just used VIsio because that's what was available to me.

**Comment by jkeck99** [comment]
Thanks for doing this.  For what it's worth, I've wasted DAYS trying to understand this.  It's painfully complex and HORRIBLY documented.  It gets 10x worse if you try to implement your own versions of the main pages.  The changes in NextAuth 5 make it ever more confusing especially in edge environments.  From what I can tell callbackUrl is depricated in V5 and they use RedirectTo
